esmvalcore/cmor/check.py

```diff
@@ -90,23 +97,55 @@ def check_metadata(self, logger=None):
             all checks and then raises.
 
         """
-        if logger is None:
-            logger = logging.getLogger(__name__)
+        if logger is not None:
+            self._logger = logger
 
         self._check_var_metadata()
         self._check_fill_value()
         self._check_dim_names()
         self._check_coords()
         if self.frequency != 'fx':
             self._check_time_coord()
+
         self._check_rank()
 
-        self.report_debug_messages(logger)
-        self.report_warnings(logger)
+        self.report_debug_messages()
+        self.report_warnings()
         self.report_errors()
 
         return self._cube
 
+    def check_data(self, logger=None):
```

Comment:
The `logger` argument is not documented, also for the other functions that have it. This class is part of the public api: https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.cmor.html#module-esmvalcore.cmor.check